### ISSUE TITLE
Hardcoded localhost for API_URL and ALLOWED_ORIGINS URL for CORS

### ISSUE THREAD

**Issue by CodingTranquility** [radualexandrub/SpringBoot-Angular-Server-Ping-Status-Tracker#2]
**Issue:** When running the application via Docker compose on a separate Server/VM, the Angular FrontEnd UI cannot be used as it's always pointing to "localhost" baseApiUrl. Also, the BackEnd Spring Boot REST API has hardcoded "localhost" URLs for setAllowedOrigins in its CORS configuration.

**Environment:** Docker compose on Redhat 8.9 VM running on Windows "Hyper-V Manager"

```bash
cat /etc/redhat-release
# Red Hat Enterprise Linux release 8.9 (Ootpa)

ip addr show eth0
# inet 172.23.100.64

sudo yum list installed | grep "docker"
# containerd.io.x86_64              1.6.28-3.1.el8  @docker-ce-stable
# docker-buildx-plugin.x86_64       0.12.1-1.el8    @docker-ce-stable
# docker-ce.x86_64                  3:25.0.3-1.el8  @docker-ce-stable
# docker-ce-cli.x86_64              1:25.0.3-1.el8  @docker-ce-stable
# docker-ce-rootless-extras.x86_64  25.0.3-1.el8    @docker-ce-stable
# docker-compose-plugin.x86_64      2.24.6-1.el8    @docker-ce-stable
```

**Expected Result:**

- Web application should retrieve the servers from backend REST API and work as expected.

**Actual Result:**

- The servers from backend REST API could not be retrieved. UI error shows:

```
Error: Error Code: 0, Message: Http failure response for http://localhost:8080/api/servers: 0 Unknown Error
```

**Notes:**

- Backend REST API http://172.23.100.64:8080/api/servers can be accessed without any issues.
- Frontend UI is at http://172.23.100.64:8081/

![CT_Hardcoded_localhost_API_URL_issue](https://github.com/radualexandrub/SpringBoot-Angular-Server-Ping-Status-Tracker/assets/68898592/de2b7d9b-5b3e-480e-87d9-eae521458e94)

**Useful commands:**

```bash
# Fix for
# ERROR [spring-api build 6/9] RUN ./mvnw dependency:go-offline -B
# > [spring-api build 6/9] RUN ./mvnw dependency:go-offline -B:
# 0.194 /bin/sh: 1: ./mvnw: Permission denied
chmod +x mvnw
```

```bash
# Revert to original code:
git stash save "Fix hardcoded API_URL and ALLOWED_ORIGINS URL for CORS"
git stash list
git stash pop
```

```bash
docker compose build angular-client
docker compose build spring-api

docker compose up -d mysql-db
docker compose logs -f mysql-db

docker compose up -d
docker compose logs -f angular-client spring-api
docker compose logs -f angular-client
docker compose logs -f spring-api
```

```bash
docker ps -a
docker compose down -v --remove-orphans
docker container prune -a 

docker image ls
docker image prune -a
# docker system prune -a
```

https://stackoverflow.com/questions/35228970/docker-compose-build-single-container

(Saturday, March 02, 2024)

**Comment by radualexandrub** [comment]
Fix added with commit be099d4 - https://github.com/radualexandrub/SpringBoot-Angular-Server-Ping-Status-Tracker/commit/be099d454cbfacc30d36babce8b96489b46f0eb5

**Comment by radualexandrub** [comment]
Fix added with commit https://github.com/radualexandrub/SpringBoot-Angular-Server-Ping-Status-Tracker/commit/be099d454cbfacc30d36babce8b96489b46f0eb5
